src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -643,7 +644,8 @@ private void CheckDisposed()
                                     break;
                                 }
                             }
-                            else if (result.IsCompleted)
+
+                            if (result.IsCompleted)
```

Comment:
Won't this throw if there was a fully parsed handshake, and the pipe was completed? I'd just get rid of the `buffer.IsEmpty` check, or don't throw if the buffer is empty.

Reply:
Or we do like the server and just return directly out of the method.